android/app/src/test/kotlin/com/vnsmap/app/routing/GraphHopperIntegrationBenchmarkTest.kt

```
@@ -0,0 +1,281 @@
+            ),
+            calculationTimeMs = 15L
+        )
+
+        val mockEngine = object : IGraphHopperEngine {
+            override fun route(fromLat: Double, fromLon: Double, toLat: Double, toLon: Double, vehicleProfile: String): RouteResult {
+                // Mô phỏng thời gian xử lý thực tế (~10ms)
+                Thread.sleep(10)
+                return urbanRouteResult
+            }
+
+            override fun close() {}
```

detekt (1.23.8)

[warning] This empty block of code can be removed.

(detekt.empty-blocks.EmptyFunctionBlock)